packaging/setup/plugins/ovirt-engine-setup/ovirt-engine-keycloak/distro-rpm/packages.py

```diff
@@ -0,0 +1,109 @@
+#
+# ovirt-engine-setup -- ovirt engine setup
+#
+# Copyright oVirt Authors
+# SPDX-License-Identifier: Apache-2.0
+#
+#
+
+
+"""
+Package upgrade plugin.
+"""
+
+import gettext
+
+
+from otopi import util
+from otopi import plugin
+
+
+from ovirt_engine_setup import constants as osetupcons
+from ovirt_engine_setup.engine import constants as oenginecons
+from ovirt_engine_setup.keycloak import constants as okkcons
+
+
+def _(m):
+    return gettext.dgettext(message=m, domain='ovirt-engine-keycloak')
+
+
+@util.export
+class Plugin(plugin.PluginBase):
+    """
+    Package upgrade plugin.
+    """
+
+    def __init__(self, context):
+        super(Plugin, self).__init__(context=context)
+
+    @plugin.event(
+        stage=plugin.Stages.STAGE_INIT,
+    )
+    def _init(self):
+        self.environment.setdefault(
+            okkcons.RPMDistroEnv.PACKAGES,
+            okkcons.Const.OVIRT_ENGINE_KEYCLOAK_PACKAGE_NAME
+        )
```

Comment:
This is not really needed anymore. Left over from times where the RHV rpm package names had 'rhevm' instead of 'ovirt-engine'. See also bz 1251129. But no harm in keeping.

Reply:
removed

Reply:
Sorry I wasn't clear. I meant that you do not need the extra flexibility of allowing the user/configuration to set the package(s) name. Now, with #8efd75c8201b065b6d344d1de3e52f5366804a5a, you keep using env[okkcons.RPMDistroEnv.PACKAGES] but do not set it anywhere.

I think you do want to add this package to lock_filter (stuff we remove from versionlock while upgrading), lock_apply (stuff we add there in the end), upgrade_list (what we upgrade), right? If you do:
- you get versionlocking and upgrading "for free"
- you risk users getting weird error messages from naive 'dnf update' when upgrades are available and some dependencies are versionlocked thus preventing updating other packages which are not versionlocked. In practice, in the few past such reports, we simply added other packages to versionlock - see e.g. bz 1404803 and bz 1321249. Not very elegant but effective. 

If you don't:
- You rely on backward/forward compatibility to work
- You allow users more choice about when to upgrade what

You might want to read a related discussion on bz 1342070.

Now pushed this, to explain what I mean, assuming you do want. Feel free to cherry-pick as-is or amend (perhaps remove/edit some comments): https://github.com/didib/ovirt-engine-keycloak/commit/510d96d6d730418928f34a48bb857d2146341ea6 .